src/components/Calculator.js

```diff
@@ -1,34 +1,69 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.clickHandler = this.clickHandler.bind(this);
+    this.displayResult = this.displayResult.bind(this);
+  }
+
+  clickHandler(e) {
+    this.setState((objName) => calculate(objName, e.target.dataset.buttonName));
+  }
+
+  displayResult() {
+    const { total, next, operation } = this.state;
+    let result = null;
+
+    if (total === null && next === null) {
+      result = 0;
+    } else if (next !== null && total === null) {
+      result = next;
+    } else if (next !== null && total !== null) {
+      if (operation !== null && operation !== undefined) {
+        result = `${total} ${operation} ${next}`;
+      } else {
+        result = total;
+      }
+    } else if (next === null && total !== null) {
+      if (operation !== null && operation !== undefined) {
+        result = `${total} ${operation}`;
+      } else {
+        result = total;
+      }
+    }
+    return result;
   }
```

Comment:
- [ ] Once I try to enter the operation before a number it display this error, kindly fix this issue by display user-friendly message to the users.

![calcc4](https://user-images.githubusercontent.com/55635977/133737743-0a7e3af1-f17f-4a4c-842c-870a290a64e7.png)

Reply:
**Changes Required**
- [ ] Good effort working on some of the requested changes. However, I still get the error when I enter the `%` operation first. Let us try to fix this as well.

https://user-images.githubusercontent.com/7220167/133748170-a3b3bd78-c6b9-4c3f-a54b-394885162c19.mp4